scorex-basics/src/main/scala/scorex/network/HistorySynchronizer.scala

```diff
@@ -89,26 +92,27 @@ class HistorySynchronizer(application: Application) extends ViewSynchronizer wit
       val localScore = history.score()
       if (networkScore > localScore) {
         log.info(s"networkScore=$networkScore > localScore=$localScore")
-        val lastIds = history.lastBlockIds(100)
+        val lastIds = history.lastBlockIds(application.blockStorage.MaxRollback)
         val msg = Message(GetSignaturesSpec, Right(lastIds), None)
         networkControllerRef ! NetworkController.SendToNetwork(msg, SendToChosen(witnesses))
-        gotoGettingExtension(networkScore, witnesses)
+        lastUpdate = Some(System.currentTimeMillis())
+        gotoGettingExtension(witnesses)
       } else gotoSynced()
   }: Receive)
 
-  def gettingExtension(betterScore: BigInt, witnesses: Seq[ConnectedPeer]): Receive = state(HistorySynchronizer.GettingExtension, {
+  def gettingExtension(witnesses: Seq[ConnectedPeer]): Receive = state(HistorySynchronizer.GettingExtension, {
     //todo: aggregating function for block ids (like score has) and blockIds type
     case DataFromPeer(msgId, blockIds: Seq[Block.BlockId]@unchecked, connectedPeer)
       if msgId == SignaturesSpec.messageCode &&
         witnesses.contains(connectedPeer) => //todo: ban if non-expected sender
 
-      lastUpdate = System.currentTimeMillis()
+      lastUpdate = Some(System.currentTimeMillis())
       val common = blockIds.head
       log.debug(s"Got blockIds: ${blockIds.map(id => Base58.encode(id))}")
 
       val toDownload = blockIds.tail.filter(b => !application.history.contains(b))
       if (application.history.contains(common) && toDownload.nonEmpty) {
-        Try(application.blockStorage.removeAfter(common)) //todo we don't need this call for blockTree
+        application.blockStorage.removeAfter(common)
```

Comment:
Yes, I understand this. My concern is this way we "eat" (ignore) all exceptions coming from removeAfter(), even those which are not related to the unsupported operation. I have changed BlockStorage.removeAfter() respectively. Maybe it makes sense in implement this in another way, but ignoring exceptions at this point is a bit risky.